packages/headless/src/controllers/commerce/product-listing/facets/headless-product-listing-facet-generator.ts

```diff
@@ -0,0 +1,57 @@
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {facetsReducer as commerceFacets} from '../../../../features/commerce/facets/facets-slice';
+import {facetOptionsReducer as facetOptions} from '../../../../features/facet-options/facet-options-slice';
+import {facetSetReducer as facetSet} from '../../../../features/facets/facet-set/facet-set-slice';
+import {
+  CommerceFacetSection,
+  FacetOptionsSection,
+  FacetSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../../controller/headless-controller';
+import {buildFacet, Facet} from './headless-product-listing-facet';
+
+export interface FacetGenerator extends Controller {
+  state: FacetGeneratorState;
+}
+
+export interface FacetGeneratorState {
+  facets: Facet[];
+}
+
+export function buildFacetGenerator(engine: CommerceEngine): FacetGenerator {
```

Comment:
This controller is very very similar to the automatic facets controller: https://github.com/coveo/ui-kit/blob/4decff2d91f415d392177b94dfd10e4c3f8566e8/packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts#L75